File: src/java/magma/app/compile/type/collect/array/ArrayType.java
package magma.app.compile.type.collect.array;

import magma.app.compile.type.Type;

public record ArrayType(Type childType) implements Type {
    @Override
    public boolean isFunctional() {
        return false;
    }

    @Override
    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "";
    }

    @Override
    public String generateSimple() {
        return "";
    }
}

File: src/ts/magma/app/compile/type/collect/array/ArrayType.ts
import { Type } from "../../../../../../magma/app/compile/type/Type";
export class ArrayType implements Type {
	childType: Type;
	constructor (childType: Type) {
		this.childType = childType;
	}
	isFunctional(): boolean {
		return false/*unknown*/;
	}
	isVar(): boolean {
		return false/*unknown*/;
	}
	generateBeforeName(): string {
		return ""/*string*/;
	}
	generateSimple(): string {
		return ""/*string*/;
	}
}
